hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java

```diff
@@ -667,6 +681,12 @@ public void handleJoinRequest(JoinRequest joinRequest, Connection connection) {
             if (now > timeToStartJoin) {
                 startJoin();
             }
+            if (node.isMaster()) {
+                long startTime = clusterClock.getClusterStartTime();
+                nodeEngine.getOperationService().invokeOnTarget(SERVICE_NAME,
```

Comment:
During join process, I tried to use FinalizeJoinOperation, MemberInfoUpdateOperation for setting clusterId or startTime, but these operations are sent to all members. What I want is, master node  sending its clusterId and startTime to new node making the join request. Besides, join process is so complicated, I did not want to touch its codebase.